### PR TITLE
FOGL-1726: first implementation od a FogLAMP plugin in its own reposi…

### PR DESCRIPTION
**Pull request by pintomax**
FogLAMP "random" C++ South plugin
  
Build
----
To build FogLAMP Random C++ South plugin:
```
- mkdir build
- cd build
- cmake ..
```

There are some options to pass to cmake:
- **FOGLAMP_SRC_DIR** sets the path of a FogLAMP source tree
- **USE_FOGLAMP_ROOT** [ON | OFF]. If set existing **FOGLAMP_ROOT** env var is set as **FOGLAMP_SRC_DIR**
- **FOGLAMP_INCLUDES** sets the path to FogLAMP header files
- **FOGLAMP_LIBS** sets the path to FogLAMP libraries

NOTE: FOGLAMP_INCLUDES must be used with FogLAMP header files all available in a single path

Exmaples:

```
cmake -DUSE_FOGLAMP_ROOT=ON ..
cmake -DFOGLAMP_SRC_DIR=/home/source/develop/FogLAMP  ..
cmake -DFOGLAMP_LIBS=/home/dev/package/lib ..
cmake -DFOGLAMP_INCLUDES=/dev-package/include ..
```

Once plugin has been built it has to be copied into an existing FogLAMP setup.

### CODE REVIEW

File: FindFogLAMP.cmake [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+# In case of error use SEND_ERROR and return()
+#
+# Remove vars from cache
+set(USE_FOGLAMP_ROOT OFF CACHE INTERNAL OFF)
+set(FOGLAMP_SRC_DIR "" CACHE INTERNAL "")
+set(FOGLAMP_DEFAULT_INCLUDE_DIR "/usr/include/foglamp" CACHE INTERNAL "")
+set(FOGLAMP_DEFAULT_LIBS_DIR "/usr/lib/foglamp" CACHE INTERNAL "")
+
+# -DUSE_FOGLAMP_ROOT=ON
+if (USE_FOGLAMP_ROOT)
```
[the remark this answers is not in the document]
Comment (pintomax) [reply FindFogLAMP.cmake]: Done

File: FindFogLAMP.cmake [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	find_path(_FIND_INCLUDES NAMES plugin_api.h PATHS ${FOGLAMP_INCLUDES})
+	if (_FIND_INCLUDES)
+		list(APPEND FOGLAMP_INCLUDE_DIRS ${_FIND_INCLUDES})
+	endif()
+	# Remove current value from cache
+	unset(_FIND_INCLUDES CACHE)
+endif()
+
+if (NOT FOGLAMP_INCLUDE_DIRS)
+	message(SEND_ERROR "FogLAMP include dir not set")
```
[the remark this answers is not in the document]
Comment (pintomax) [reply FindFogLAMP.cmake]: Done

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
+FogLAMP "random" C++ South plugin
+
+Build
+----
+To build FogLAMP Random C++ South plugin:
+- mkdir build
+- cd build
+- cmake ..
+
+There are some options to pass to cmake:
```
[the remark this answers is not in the document]
Comment (pintomax) [reply README.rst]: Done

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
+FogLAMP "random" C++ South plugin
+
+Build
+----
+To build FogLAMP Random C++ South plugin:
+- mkdir build
+- cd build
+- cmake ..
+
+- By default the FogLAMP develop package header files an libraries
```
Comment (MarkRiddoch) [inline README.rst]: typo **and** libraries

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  are in use: /usr/include/foglamp and /usr/lib/foglamp
```
Comment (MarkRiddoch) [inline README.rst]: "in use" -> "expected to be located in"

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- If FOGLAMP_ROOT env var is set and no -D options are set,
+  the header files and libraries paths are set to FOGLAMP_ROOT dir
```
Comment (MarkRiddoch) [inline README.rst]: "are set to FOGLAMP_ROOT dir" -> "are pulled from the ones under the FOGLAMP_ROOT directory. Please note that you must first do a make in the FOGLAMP_ROOT directory."

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+Options to pass to cmake:
```
Comment (MarkRiddoch) [inline README.rst]: "Options to pass to cmake" -> "You may also pass one or more of the following options to cmake to override this default behaviour."

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+- FOGLAMP_SRC_DIR sets the path of a FogLAMP source tree
+- FOGLAMP_INCLUDES sets the path to FogLAMP header files
+- FOGLAMP_LIBS sets the path to FogLAMP libraries
+- FOGLAMP_INSTALL_DIR sets the installation path of Random plugin
+
+NOTE:
+ - FOGLAMP_INCLUDES must be used with FogLAMP header files all available in a single path
```
Comment (MarkRiddoch) [inline README.rst]: "FOGLAMP_INCLUDES must be used with FogLAMP header files all available in a single path" ->
"The FOGLAMP_INCLUDES option should point to a location where all the FogLAMP header files have be installed in a single directory."

You also need to same something similar about FOGLAMP_LIBS.

Slightly inconsistency naming here as well, some have _DIR on the end some don't. Some are singular and other are plural. I would be tempted to drop the _DIR part and make them all singular.

**Pull request merged**